lib/spoom/location.rb

```diff
@@ -15,42 +15,75 @@ class << self
       sig { params(location_string: String).returns(Location) }
       def from_string(location_string)
         file, rest = location_string.split(":", 2)
-        raise LocationError, "Invalid location string: #{location_string}" unless file && rest
+        raise LocationError, "Invalid location string `#{location_string}`: missing file name" unless file
+
+        return new(file) if rest.nil?
 
         start_line, rest = rest.split(":", 2)
-        raise LocationError, "Invalid location string: #{location_string}" unless start_line && rest
+        if rest.nil?
+          start_line, end_line = T.must(start_line).split("-", 2)
+          raise LocationError, "Invalid location string `#{location_string}`: missing end line" unless end_line
+
+          return new(file, start_line: start_line.to_i, end_line: end_line.to_i) if end_line
+        end
 
         start_column, rest = rest.split("-", 2)
-        raise LocationError, "Invalid location string: #{location_string}" unless start_column && rest
+        if rest.nil?
+          raise LocationError, "Invalid location string `#{location_string}`: missing end line and column"
```

Comment:
Does this fit in one line?
```suggestion
        raise LocationError, "Invalid location string `#{location_string}`: missing end line and column" if rest.nil?
```

Reply:
It does, good catch! 🕵️